core/src/main/java/com/linecorp/armeria/client/HttpClient.java

```diff
@@ -139,6 +139,52 @@ static HttpClient of(ClientFactory factory, URI uri, ClientOptions options) {
         return new HttpClientBuilder(uri).factory(factory).options(options).build();
     }
 
+    /**
+     * Creates a new HTTP client that connects to the specified {@link Endpoint} using the default
+     * {@link ClientFactory}.
+     *
+     * @param endpoint the URI of the server endpoint {@link Endpoint}
+     * @param options the {@link ClientOptionValue}s
+     */
+    static HttpClient of(Endpoint endpoint, ClientOptionValue<?>... options) {
+        return of(ClientFactory.DEFAULT, endpoint.toURI(), options);
```

Comment:
`requireNonNull(endpoint, "endpoint")`
Should check if the `endpoint` is `null` before calling its method.